docs/dev/how-to/verify-contracts.md

```diff
@@ -0,0 +1,283 @@
+const zkSyncTestnet =
+  process.env.NODE_ENV == "test"
+    ? {
+        url: "http://localhost:3050",
+        ethNetwork: "http://localhost:8545",
+        zksync: true,
+      }
+    : {
+        url: "https://zksync2-testnet.zksync.dev",
+        ethNetwork: "goerli",
+        zksync: true,
+        verifyURL: 'https://zksync2-testnet-explorer.zksync.dev/contract_verification'  // Verification endpoint
+      };
+
+const config: HardhatUserConfig = {
+  zksolc: {
+    version: "1.3.10",
+    compilerSource: "binary",
+    settings: {},
+  },
+  defaultNetwork: "zkSyncTestnet",
+  networks: {
+    hardhat: {
+      zksync: false,
+    },
+    zkSyncTestnet,
+  },
+  // etherscan: { // Optional - If you plan on verifying a smart contract on Ethereum within the same project
+  //   apiKey: //<Your API key for Etherscan>,
+  // },
+  solidity: {
+    version: "0.8.17",
+  },
+};
+
+export default config;
+```
+
+- Feel free to assign an arbitrary zkSync Era network name utilizing the `defaultNetwork` property. 
+- The `verifyURL` attribute directs to the verification endpoint specific to the zkSync network.
+- If you intend to validate a smart contract on Ethereum within the same project, don't forget to include your [Etherscan API key](https://docs.etherscan.io/getting-started/viewing-api-usage-statistics).
+
+### 4. Greeter Contract Compilation
+
+The [zkSync CLI](/docs/tools/zksync-cli/) provides a `Greeter.sol` contract we will verify on zkSync Era. 
+
+Compile the contract using this command:
+
+```sh
+yarn hardhat compile
+```
+
+### 5. Deployment of the Greeter Contract 
+
+The [zkSync CLI](/docs/tools/zksync-cli/) provides a `deploy/deploy-greeter.ts` script that we will use to deploy the Greeter contract. Replace `<WALLET-PRIVATE-KEY>` with your key.
+
+Initiate contract deployment using this command:
+
+```sh
+yarn hardhat deploy-zksync --script deploy-greeter.ts
+```
+
+Expect an output similar to this:
+
+```txt
+Running the deployment function for the Greeter contract
+The deployment is estimated to cost 0.0265726735 ETH
+constructor args:0x000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000094869207468657265210000000000000000000000000000000000000000000000
+The Greeter contract got deployed at 0xE84774C41F096Ba5BafA1439cEE787D9dD1A6b72
+```
+
+Remember, you will need the contract address to verify the contract on zkSync Era. 
+
+### 6. Verify the contract
+
+Run the following command to verify your contract on the specified network, replacing `<contract address>` with your contract's actual address.
+
+```sh
+yarn hardhat verify --network <network> <contract address>
+```
+
+For example, to verify our Greeter contract on the `zkSyncTestnet` network and our contract address is `0xE84774C41F096Ba5BafA1439cEE787D9dD1A6b72`, use:
+
+```sh
+yarn hardhat verify --network zkSyncTestnet 0xE84774C41F096Ba5BafA1439cEE787D9dD1A6b72 'Hi There!'
+```
```

Comment:
I get errors here because there is no "Hi There!" in the deploy, I'm guessing. 

With Hi there, copy/pasted, I get:

```
Error in plugin @matterlabs/hardhat-zksync-verify: Constructor arguments are not correct
```

With no constructor I get:

```
Error in plugin @matterlabs/hardhat-zksync-verify: The number of constructor arguments you provided (0) does not match the number of constructor arguments the contract has been deployed with (1).
```

Reply:
My mistake, the args in the docs should be 'Hi there!' instead of 'Hi There!'. Updated in latest commit.